src/rail/estimation/algos/deepdisc.py

```diff
@@ -143,11 +145,41 @@ class DeepDiscInformer(CatInformer):
     def __init__(self, args, comm=None):
         CatInformer.__init__(self, args, comm=comm)
 
+    def inform(self, training_data, metadata):
+        self.set_data('input', training_data)
+        self.set_data('metadata', metadata)
+        self.run()
+        self.finalize()
+        return self.get_handle('model')
+
+    def finalize(self):
+        self.temp_dir.cleanup()
+
     def run(self):
         """
         Train a inception NN on a fraction of the training data
         """
         train_data = self.get_data("input")
+        metadata = self.get_data("metadata")
+
+        # create an iterator here
+        itr = self.input_iterator("input")
+        self.temp_dir = tempfile.TemporaryDirectory()
+        for start_idx, _, chunk in itr:
+            for idx, image in enumerate(chunk):
```

Comment:
We should probably move this nested for loop into a distinct function so that it can be reused (if needed) or at least more easily tested in isolation.

Reply:
I don't think we'd necessarily need to reuse it, but it may be nice to package this in a distinct class method

Reply:
Tracked in https://github.com/lincc-frameworks/deepdisc/issues/76